examples/nextjs-flow/app/hooks/newPartials.ts

```diff
@@ -0,0 +1,369 @@
+// types.ts
+
+import type { FlowClientIncomingMessage } from '@speechmatics/flow-client-react';
+
+/**
+ * Represents incoming messages from the speech-to-text service
+ * These messages can be transcripts, partial transcripts, or agent responses
+ */
+export interface FlowMessage {
+  message:
+    | 'AddPartialTranscript'
+    | 'AddTranscript'
+    | 'ResponseStarted'
+    | 'ResponseCompleted'
+    | 'ResponseInterrupted';
+  results?: Array<{
+    start_time: number;
+    end_time: number;
+    type?: 'punctuation';
+    is_eos?: boolean; // end of sentence marker
+    attaches_to?: 'previous' | 'next'; // indicates how punctuation attaches to words
+    alternatives?: Array<{
+      content: string; // the actual text content
+      speaker: string; // speaker identifier
+    }>;
+  }>;
+  start_time?: number;
+  end_time?: number;
+  content?: string;
+}
+
+/**
+ * Represents a single word in the transcript
+ * Can be either a regular word or punctuation
+ */
+export type Word = {
+  startTime: number;
+  endTime: number;
+  speaker: string;
+  text: string;
+  partial: boolean; // indicates if this is from a partial transcript
+} & (
+  | { punctuation: false }
+  | {
+      punctuation: true;
+      eos: boolean; // end of sentence marker
+      attachesTo?: 'previous' | 'next';
+    }
+);
+
+/**
+ * Represents a response from an AI agent in the conversation
+ */
+export type AgentResponse = {
+  speaker: 'agent';
+  agent: true;
+  startTime: number;
+  endTime?: number;
+  text: string;
+};
+
+/**
+ * Groups of transcript data, either from a human speaker or an AI agent
+ */
+export type TranscriptGroup =
+  | {
+      type: 'speaker';
+      speaker: string;
+      data: Word[];
+    }
+  | {
+      type: 'agent';
+      data: AgentResponse[];
+    };
+
+// transcript-manager.ts
+/**
+ * Manages the state and processing of a real-time transcript
+ * Handles both human speech transcription and AI agent responses
+ * Uses EventTarget to notify listeners of updates
+ */
+export class TranscriptManager extends EventTarget {
```

Comment:
I'd recommend using the `TypedEventTarget` from the `typescript-event-target` class for this (it's already a dependency I think). Benefit is you can strictly type the event map, so consumers don't have to do type checks